src/compiler/scala/tools/nsc/backend/icode/GenICode.scala

```diff
@@ -876,13 +876,15 @@ abstract class GenICode extends SubComponent {
                 genLoadModule(ctx, tree)
                 generatedType = toTypeKind(sym.info)
               } else {
-                try {
-                  val Some(l) = ctx.method.lookupLocal(sym)
-                  ctx.bb.emit(LOAD_LOCAL(l), tree.pos)
-                  generatedType = l.kind
-                } catch {
-                  case ex: MatchError =>
-                    abort("symbol " + sym + " does not exist in " + ctx.method)
+                ctx.method.lookupLocal(sym) match {
+                  case Some(l) =>
```

Comment:
It sure would be nice if that didn't look like a one. It's too bad I sound so ungrateful when I say that. I bet we could make good use of the Turkish undotted i.